west.yml

```diff
@@ -21,6 +21,8 @@ manifest:
   remotes:
     - name: upstream
       url-base: https://github.com/zephyrproject-rtos
+    - name: meekrosoft
+      url-base: https://github.com/meekrosoft
```

Comment:
Why can't we use the upstream (assuming they accept the PR to include `zephyr/module.yml`)? We're fixing the SHA so even if they add a breaking change we would need to uprev it locally and test it so it's not going to break anything.

Reply:
All modules must have a fork in Zephyr.
Please read the guide-lines:
https://docs.zephyrproject.org/latest/contribute/external.html#integration-as-a-module
https://docs.zephyrproject.org/latest/contribute/external.html#external-src-process
https://docs.zephyrproject.org/latest/guides/modules.html#process-for-submitting-a-new-module

Reply:
I read those very carefully, the only mention of a fork is:

> If a module is maintained as a fork of another project on Github, the Zephyr module related files and changes in relation to upstream need to be maintained in a special branch named `zephyr`.

There's no hard requirement for it to be a fork neither from the documentation nor from a technical perspective. West is perfectly capable of handling more than one remote and instead of cluttering the `zephyrproject-rtos` with forks I would highly recommend we make use of this and first try to get upstream support in our modules. We can further discuss in this week's TSC meeting if you'd like.

Reply:
This has nothing to do with `west`.

It's an old decision that Zephyr project wants to be in control of all projects in the manifest.
Otherwise Zephyr bears the risk of a repo suddenly missing, a lost SHA (force push to upstream with history rewrite) or the project changing its license (in this case Zephyr can decide to stay at the version with an acceptable license).

Also note, for example a Zephyr LTS will be supported for long period, in which case there is an increased risk when depending on external repo.

With a fork, Zephyr community decides when to update, and in case something changes, whether or not to upmerge those changes.

Reply:
Those are valid concerns, can we update the documentation to reflect that? I wouldn't trust myself to get the wording right.

Reply:
> Those are valid concerns, can we update the documentation to reflect that? I wouldn't trust myself to get the wording right.

it should probably be documented (if it's not done so somewhere else).
But I don't trust my own wording on this either.

Reply:
it seems to be documented here:
https://docs.zephyrproject.org/latest/guides/modules.html#module-repositories

> Module Repositories
>
>   All modules included in the default manifest shall be hosted in repositories under the zephyrproject-rtos GitHub organization.

on the same page I linked earlier.

Reply:
I see, I was looking for the word `fork`, must have missed this. I'll keep working on the tests to make them cleaner and avoid having to copy them over (making good progress). Hopefully they'll be accepted upstream, if not then I'll just send them to the fork.

Reply:
With the latest changes I'm very happy with this PR as it's able to run all of the C tests for FFF. @carlescufi what are your thoughts about creating the fork? We can merge my FFF PR there while it's being reviewed upstream and start using this. I want to make a big push for testing on Zephyr and this is a key aspect of it.